test/validation-error-handling.test.js

```diff
@@ -831,3 +831,52 @@ test('plugin override', async (t) => {
   })
   t.assert.strictEqual(response5.statusCode, 400)
 })
+
+test('async preValidation with custom validator should trigger error handler when validator throws', async (t) => {
+  t.plan(4)
+
+  const fastify = Fastify()
+
+  // Set up error handler
+  fastify.setErrorHandler((error, request, reply) => {
+    t.assert.ok(error instanceof Error, 'error should be an Error instance')
+    t.assert.strictEqual(error.message, 'Custom validation failed')
+    reply.status(400).send({ error: error.message })
```

Comment:
So we can test that the status code is correctly set

```suggestion
    reply.send({ error: error.message })
```